core/src/main/java/org/apache/spark/shuffle/api/ShuffleExecutorComponents.java

```diff
@@ -39,16 +39,20 @@
   /**
    * Called once per map task to create a writer that will be responsible for persisting all the
    * partitioned bytes written by that map task.
-   *  @param shuffleId Unique identifier for the shuffle the map task is a part of
+   * @param shuffleId Unique identifier for the shuffle the map task is a part of
+   * @param shuffleGenerationId The shuffle generation ID of the stage that this task belongs to,
```

Comment:
Why can't we just use the `mapTaskAttemptId`? (In fact I wonder if we can just have a global identifier and not even need `shuffleId` and `mapId` and just use `mapTaskAttemptId` as a global identifier, but that might be a bit ambitious.)

Reply:
Specifically taking a look at the linked PR for indeterminate retries - I'd expect that on a rolled back map stage, the implementation of this plugin will be given a different `mapTaskAttemptId` anyways since that's going to be updated on the resubmit. So, we'll end up opening a new writer regardless, but, we could have gotten the same behavior just by using the `mapTaskAttemptId`.

Reply:
Actually let's move discussion over to #24892

Reply:
Yep, thanks for your faster review, I describe the requirement in https://github.com/apache/spark/pull/24892#issuecomment-518487347